distributed/shuffle/tests/test_shuffle.py

```diff
@@ -2057,6 +2058,28 @@ async def test_handle_null_partitions_p2p_shuffling(c, s, *workers):
     await check_scheduler_cleanup(s)
 
 
+@gen_cluster(client=True)
+async def test_handle_null_partitions_p2p_shuffling_2(c, s, a, b):
+    def make_partition(i):
+        """Return null column for one partition"""
+        if i % 2 == 1:
+            return pd.DataFrame({"a": np.random.random(10), "b": [None] * 10})
```

Comment:
I would make the None column null explicitly, otherwise lgtm

Reply:
This fails unless I replace `b` in the other case with a string. That's another issue I plan to handle. As the current test mimics the original reproducer, I will keep this.